src/h5web/dataset-visualizer/DatasetVisualizer.tsx

```diff
@@ -66,30 +29,20 @@ function DatasetVisualizer(props: Props): ReactElement {
         <VisSelector
           activeVis={activeVis}
           choices={supportedVis}
-          onChange={(vis) => {
-            mergeState({
-              activeVis: vis,
-              mapperState: VIS_DEFS[vis].defaultMappingState(dataset),
-            });
-          }}
+          onChange={setActiveVis}
         />
         {VisToolbar && <VisToolbar />}
       </div>
       <div className={styles.displayArea}>
         <VisDisplay
+          key={`${dataset.id}_${activeVis}`}
```

Comment:
This forces a reset of the state of `VisDisplay` (dataset value + mapper state) whenever the visualization and/or dataset changes.

Reply:
I would rather see this in comment in the code :wink;

Reply:
Oh sorry, I did mean to 😓